qiskit/circuit/qpy_serialization.py

```diff
@@ -25,3 +27,25 @@
     _read_parameter_expression,
     _read_parameter_expression_v3,
 )
+
+deprecated_names = {
+    "dump",
+    "load",
+    "_write_instruction",
+    "_read_instruction",
+    "_write_parameter_expression",
+    "_write_parameter_expression",
+}
+
+
+def __getattr__(name):
+    if name in deprecated_names:
+        warnings.warn(
+            "The qiskit.circuit.qpy_serialization module is deprecated and has been supersceded "
+            f"by the qiskit.qpy module. The {name} function should be accessed from the qiskit.qpy "
+            "module instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        return globals()[name]
+    raise AttributeError(f"module {__name__} has no attribute {name}")
```

Comment:
I don't think this actually shows the deprecation warnings, because `__getattr__` is only called if standard lookups fail.  Since `dump` and friends are available in the regular module dictionary, I don't think this will trigger.

Reply:
Good catch, fixed in: https://github.com/Qiskit/qiskit-terra/pull/8023/commits/616fcf353fa201f5224a83bacf3a4cce697e0a1f